basics/connecting_to_databases.md

```diff
@@ -0,0 +1,298 @@
+
+### Connect to SQLite with the FFI Module
+
+[sqlite3](https://deno.land/x/sqlite3) provides JavaScript bindings to the
+SQLite3 C API, using Deno FFI.
+
+```ts, ignore
+import { Database } from "https://deno.land/x/sqlite3@LATEST_VERSION/mod.ts";
+
+const db = new Database("test.db");
+
+db.close();
+```
+
+### Connect to SQLite with the WASM-Optimized Module
+
+[sqlite](https://deno.land/x/sqlite) is a SQLite module for JavaScript and
+TypeScript. The wrapper is targeted at Deno and uses a version of SQLite3
+compiled to WebAssembly (WASM).
+
+```ts, ignore
+import { DB } from "https://deno.land/x/sqlite/mod.ts";
+
+const db = new DB("test.db");
+
+db.close();
+```
+
+## Firebase
+
+To connect to Firebase with Deno, import the
+[firestore npm module](https://firebase.google.com/docs/firestore/quickstart)
+with the [skypak CDN](https://www.skypack.dev/). To learn more about using npm
+modules in Deno via CDN read [here](../basics/node/cdns.md)
+
+### Connect to Firebase with the firestore npm module
+
+```js, ignore
+import { initializeApp } from "https://www.gstatic.com/firebasejs/9.8.1/firebase-app.js";
+
+import {
+  addDoc,
+  collection,
+  connectFirestoreEmulator,
+  deleteDoc,
+  doc,
+  Firestore,
+  getDoc,
+  getDocs,
+  getFirestore,
+  query,
+  QuerySnapshot,
+  setDoc,
+  where,
+} from "https://www.gstatic.com/firebasejs/9.8.1/firebase-firestore.js";
+
+import { getAuth } from "https://www.gstatic.com/firebasejs/9.8.1/firebase-auth.js";
+
+const app = initializeApp({
+  apiKey: Deno.env.get("FIREBASE_API_KEY"),
+  authDomain: Deno.env.get("FIREBASE_AUTH_DOMAIN"),
+  projectId: Deno.env.get("FIREBASE_PROJECT_ID"),
+  storageBucket: Deno.env.get("FIREBASE_STORAGE_BUCKET"),
+  messagingSenderId: Deno.env.get("FIREBASE_MESSING_SENDER_ID"),
+  appId: Deno.env.get("FIREBASE_APP_ID"),
+  measurementId: Deno.env.get("FIREBASE_MEASUREMENT_ID"),
+});
+const db = getFirestore(app);
+const auth = getAuth(app);
+```
+
+## Supabase
+
+To connect to Supabase with Deno, import the
+[supabase-js npm module](https://supabase.com/docs/reference/javascript) with
+the [esm.sh CDN](https://esm.sh/). To learn more about using npm modules in Deno
+via CDN read [here](../basics/node/cdns.md)
+
+### Connect to Supabase with the supabase-js npm module
+
+```js, ignore
+import { createClient } from "https://esm.sh/@supabase/supabase-js";
+
+const options = {
+  schema: "public",
+  headers: { "x-my-custom-header": "my-app-name" },
+  autoRefreshToken: true,
+  persistSession: true,
+  detectSessionInUrl: true,
+};
+
+const supabase = createClient(
+  "https://xyzcompany.supabase.co",
+  "public-anon-key",
+  options,
+);
+```
+
+## ORMs
+
+The idea with ORMs (object relational mapping) is to define your data models as
+classes that can be persisted to a database. The classes and their instances
+then provide you with a programmatic API to read and write data in the database.
+
+Deno has support for a number of popular ORMs, including Prisma and DenoDB.
```

Comment:
No specific information on Prisma?

Reply:
this is what you guys said earlier:

_prisma - this the very popular one at the moment, but I don't know where we stand on Deno compatibility

if it's indeed compatible with Deno, can you send me a link to the appropriate third party module/driver?